spec/requests/api/generic_events_controller_create_spec.rb

```diff
@@ -93,13 +93,67 @@
       expect { do_post }.to change { GenericEvent.find_by(eventable_id: move.id, eventable_type: 'Move') }.from(nil).to(be_a(GenericEvent))
     end
 
-    it 'returns serialized data' do
-      do_post
+    context 'when the event has no additional relationships' do
+      let(:event_attributes) do
+        attributes_for(:event_move_cancel).tap do |attributes|
+          attributes.except!(:eventable)
+          attributes[:event_type] = 'MoveCancel'
+        end
+      end
+
+      it 'returns serialized data' do
+        do_post
+
+        event = GenericEvent.last
+        resource_to_json = JSON.parse(event.class.serializer.new(event).serializable_hash.to_json)
+
+        expect(response_json).to eq resource_to_json
+      end
+
+      it 'returns the serialized relationships of the event' do
+        do_post
 
-      event = GenericEvent.last
-      resource_to_json = JSON.parse(GenericEventSerializer.new(event).serializable_hash.to_json)
+        event = GenericEvent.last
+        resource_to_json = JSON.parse(event.class.serializer.new(event).serializable_hash.to_json)
+
+        expected_relationships = {
+          'eventable' => { 'data' => { 'id' => event.eventable.id, 'type' => 'moves' } },
+        }
 
-      expect(response_json).to eq resource_to_json
+        expect(resource_to_json.dig('data', 'relationships')).to eq(expected_relationships)
+      end
+    end
+
+    context 'when the event has an additional relationship' do
+      let(:event_attributes) do
+        attributes_for(:event_move_lockout).tap do |attributes|
+          attributes.except!(:eventable)
+          attributes[:details] = attributes[:details].except(:from_location_id)
+          attributes[:event_type] = 'MoveLockout'
```

Comment:
This is starting to require a little bit of mental gymnastics to figure out what the expected attributes will be - perhaps clearer to make this explicit even if it means repeating some of the attributes again.

Reply:
Fixed by being explicit about attributes.